WordPress/Classes/Extensions/NSString+Helpers.swift

```diff
@@ -19,4 +25,16 @@ extension NSString
         
         return uniqueSet
     }
+    
+    /// Validates the current string.
+    ///
+    /// -   returns: true if passes validation
+    ///
+    public func isValidEmail() -> Bool {
+        // From http://stackoverflow.com/a/3638271/1379066
+        let emailRegex = ".+@([A-Za-z0-9-]+\\.)+[A-Za-z]{2}[A-Za-z]*"
```

Comment:
I'm very wary of checking emails with regexps, as it can be a source of bugs and edge cases, and nothing beats actually sending an email to the address to validate it. If you want a quick check just check that there's a `@` in the string.

As an example, this regex fails to validate all of these:

```
// Although rare, TLDs can have email too
"koke@com".isValidEmail()
// ...the hyphen is permitted if it is surrounded by characters, digits or hyphens,
// although it is not to start or end a label
"koke@-example.com".isValidEmail()
"koke@example-.com".isValidEmail()
// https://en.wikipedia.org/wiki/International_email
"用户@例子.广告".isValidEmail()
"उपयोगकर्ता@उदाहरण.कॉम".isValidEmail()
```